File: 1-TwoSum/1-TwoSum.py
# Last updated: 09/10/2025, 09:40:35
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}  # num -> index

        for i, num in enumerate(nums):
            diff = target - num
            if diff in seen:
                return [seen[diff], i]
            seen[num] = i

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 01/08/2026, 23:26:06
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        int n = nums.size();
        vector <int> res;
        for(int i = 0; i< n; i++)
        {
            int x = target - nums[i];

            for( int j = i+1; j<n; j++)
            {
                if(nums[j] == x)
                {
                    res.push_back(i);
                    res.push_back(j);
                }
            }
        }

         return res;
        
    }
};